node/native.js

```diff
@@ -14,12 +14,26 @@
 
 let nativeLib;
 
-if (process.platform === "darwin" && process.arch === "arm64") {
-    nativeLib = require('./darwin_arm64.node')
-} else if (process.platform === "linux" && process.arch === "x64") {
-    nativeLib = require('./linux-x64.node')
-} else {
-    throw new Error(`vectordb: unsupported platform ${process.platform}_${process.arch}. Please file a bug report at https://github.com/lancedb/lancedb/issues`)
+function getPlatformLibrary() {
```

Comment:
I'll fill this out in the next PR, this is mostly a placeholder.

Reply:
For releases, that's still to be decided. But for local development, I think you'll just build index.node for your local platform and it will pick that up. In release it will instead download the the right platform build named `<platform>.node`, and that's what this function is starting to handle. But that will be filled in the next PR.